File: app/src/main/java/me/rohank05/beat/command/commands/PauseCommand.java
package me.rohank05.beat.command.commands;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import me.rohank05.beat.command.ICommand;
import me.rohank05.beat.lavaplayer.PlayerManager;

import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;

import java.awt.Color;


public class PauseCommand implements ICommand {
    @Override
    public void run(SlashCommandInteractionEvent event) {
        if (!CheckMusicPermission.checkRemainPermission(event)) return;
        if (!CheckMusicPermission.checkPermission(event)) return;
        AudioPlayer audioPlayer = PlayerManager.getINSTANCE().getMusicManager(event.getGuild()).audioPlayer;
        EmbedBuilder embed = new EmbedBuilder();
        if (audioPlayer.isPaused()) {
            event.getInteraction().getHook().sendMessageEmbeds(embed.setColor(Color.RED).setDescription("Track is already paused").build());
        } else {
            audioPlayer.setPaused(true);
            event.getInteraction().getHook().sendMessageEmbeds(embed.setColor(Color.RED).setDescription("Track Paused").build());
        }
    }

    @Override
    public String getName() {
        return "pause";
    }
}

File: src/main/kotlin/com/therohankumar/commands/PauseCommand.kt
package com.therohankumar.commands

import com.therohankumar.audio.MusicManagerRegistry
import com.therohankumar.commands.util.CommandUtil
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.build.Commands
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData
import org.springframework.stereotype.Component

@Component
class PauseCommand(private val registry: MusicManagerRegistry) : SlashCommand {

    override val name = "pause"

    override fun build(): SlashCommandData = Commands.slash(name, "Pause the current track")

    override fun execute(event: SlashCommandInteractionEvent) {
        val manager = CommandUtil.requireSameVoice(event, registry) ?: return
        manager.player.isPaused = true
        event.replyEmbeds(CommandUtil.successEmbed("Paused", "Use `/resume` to continue.")).queue()
    }
}
